unitypackage_extractor/extractor.py

```diff
@@ -29,9 +30,14 @@ def extractPackage(packagePath, outputPath="", encoding='utf-8'):
         pathname = f.readline()
         pathname = pathname[:-1] if pathname[-1] == '\n' else pathname #Remove newline
 
+      # Figure out final path, make sure that it's inside the write directory
+      assetOutPath = os.path.join(outputPath, pathname)
+      if Path(outputPath) not in Path(assetOutPath).resolve().parents:
```

Comment:
I don't know if I'm missing something but this seems like it'll only work if outputPath is absolute. Path comparisons seem to work like string comparisons, so relative paths != absolute paths even if they're referring to the same place:

```python
>>> Path.cwd() == Path(".")
False
```

I think you need to resolve the path on the left, first, or else this will always fail on relative paths.

Reply:
I tested this and it really doesn't behave as expected because of the above and the fact that Path.resolve() requires paths to be existent. I submitted #16 that uses os.path.abspath() instead, if you merge it it should be merged into this PR.

Reply:
Yeah, my mistake on the first part. Both need to be resolved, otherwise `.parents` won't actually match anything.

Reply:
I'm not seeing what you mean for the second part though? It seems to leave non-existent folder's in, and properly handles relative parts in the non-existent portion.

```python
>>> Path("./dist/asdf/../jjjj/jkke/..").resolve()
PosixPath('/home/cobertos/Seafile/projects/unitypackage_extractor/dist/jjjj')
>>> os.path.abspath("./dist/asdf/../jjjj/jkke/..")
'/home/cobertos/Seafile/projects/unitypackage_extractor/dist/jjjj'
```

Reply:
Added another test as well to test the `resolve()` issue, by specifying output as a relative path

Reply:
It won't on Windows which is where I've tested it (Python 3.9.6). Yes, this means the Python pathlib docs are wrong. On Windows, if the path doesn't exist, it won't turn it into an absolute path. That breaks `Path.parents`, unsurprisingly, among other stuff (which led to the issues I described)

```python
>>> Path("./example").resolve()
WindowsPath('example')
>>> Path("..").resolve()
WindowsPath('C:/Users')
```

Reply:
Ahh, figured it might've been a platform specific thing. Why does the Windows implementation do this.

I will try this on my Windows machine when I finish getting food. Thanks for the follow up

Reply:
Wow yeah, this is really annoying... A hidden footgun in pathlib, cool... And no good way to get an absolute path of a `PurePath` either.